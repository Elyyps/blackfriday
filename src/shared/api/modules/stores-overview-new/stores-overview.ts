import { WordPressPostModule } from "../wordpress-module/wordpress-module";
import { Filter } from "@app/api/core/filter/filter";

export class StoresOverviewModule extends WordPressPostModule {
  public filters: Filter[];
  public totalShops: number;
}
