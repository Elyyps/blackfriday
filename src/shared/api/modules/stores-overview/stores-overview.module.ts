import { ShopCard } from "@app/api/core/shop-card/shop-card";
import { FilterBar } from "@app/api/core/filter-bar/filter-bar";
import { generateShopCardData, generateShopCardData2 } from "@app/api/core/shop-card/endpoint";
import { WordPressPostModule } from "../wordpress-module/wordpress-module";

export class StoresOverviewModule extends WordPressPostModule {
  public filterBar: FilterBar;
  public shopCards: ShopCard[];
  public totalCards: number;
}
export const shuffleArray = (a: any) => {
  const result = [...a];

  for (let i = 1; result.length > i; i++) {
    const j = Math.floor(Math.random() * (i + 1));
    [result[i], result[j]] = [result[j], result[i]];
  }

  return result;
};

export function getShopCards(
  currentPage: number,
  status: string[],
  categories: string[],
  brands: string[],
  sortBy: string
) {
  const shopCards: ShopCard[] = generateShopCardData2();
  let filteredCardsByStatus: ShopCard[] = [];
  let filteredCardsByCategories: ShopCard[] = [];
  let filteredCardsByBrand: ShopCard[] = [];
  let SortedCards: ShopCard[] = [];
  const TAKE = 15;
  const skip: number = TAKE * currentPage;
  const numberOfCards = skip + TAKE;

  if (status.length >= 1) {
    shopCards.map(card => status.includes(card.timeLeftBar.text.toUpperCase()) && filteredCardsByStatus.push(card));
  } else {
    filteredCardsByStatus = shopCards;
  }
  if (categories.length >= 1) {
    filteredCardsByStatus.map(
      card => categories.includes(card.category.toUpperCase()) && filteredCardsByCategories.push(card)
    );
  } else {
    filteredCardsByCategories = filteredCardsByStatus;
  }
  if (brands.length >= 1) {
    filteredCardsByCategories.map(card => brands.includes(card.brand.toUpperCase()) && filteredCardsByBrand.push(card));
  } else {
    filteredCardsByBrand = filteredCardsByCategories;
  }
  if (sortBy.length >= 1) {
    sortBy === "Newest"
      ? (SortedCards = shuffleArray(filteredCardsByBrand))
      : sortBy === "Price"
      ? (SortedCards = shuffleArray(filteredCardsByBrand))
      : sortBy === "Relevant" && (SortedCards = shuffleArray(filteredCardsByBrand));
  } else {
    SortedCards = filteredCardsByBrand;
  }
  SortedCards = SortedCards.slice(skip, numberOfCards);
  // console.log(skip + " skip");
  // console.log(numberOfCards + " numberOfCards");
  console.log(currentPage + " currentPage");

  // console.log(SortedCards.length);

  return { shopCards: SortedCards, totalCards: SortedCards.length };
}
