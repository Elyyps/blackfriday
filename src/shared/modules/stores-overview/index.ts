export * from "./stores-overview.component";
