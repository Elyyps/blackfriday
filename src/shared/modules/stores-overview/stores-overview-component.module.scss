.stores-overview {
  background-color: var(--tint-grey-97);
  &__header {
    padding-bottom: 16px;
    background-color: var(--tint-white);
  }
  &__body {
    width: 100%;
    margin: auto;
    &__list {
      flex-wrap: wrap;
      flex-direction: row;
      display: flex;
    }
    &__cards {
      flex: 0 1 19%;
      padding: 4px;
      @media (max-width: 1160px) {
        flex: 0 1 24%;
      }
      @media (max-width: 950px) {
        padding: 3px;
        flex: 0 1 32.5%;
      }
      @media (max-width: 770px) {
        padding: 1.5px;
      }
      @media (max-width: 699px) {
        flex: 0 1 100%;
      }
    }
  }
}
