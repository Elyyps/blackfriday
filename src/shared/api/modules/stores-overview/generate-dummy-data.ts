import { StoresOverviewModule } from "./stores-overview.module";
import { generateData, generateDummyShopCards } from "@app/api/core/shop-card";
import { generateDummyFilterBar } from "@app/api/core/filter-bar";

export const generateDummyStoresOverview = (): StoresOverviewModule => ({
  shopCards: generateDummyShopCards(),
  filterBar: generateDummyFilterBar(),
  totalCards: 1
});
