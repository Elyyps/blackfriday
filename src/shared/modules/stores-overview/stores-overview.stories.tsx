import * as React from "react";
import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { StoresOverviewComponent } from "./stores-overview.component";
import {
  generateShopsOverviewData,
  generateEmptyShopsOverviewData,
  generateShopsOverviewDataForInfiniteScrolling
} from "@app/api/modules/stores-overview/endpoints";
const storesOverviewData = generateShopsOverviewData();
const storesOverviewEmptyData = generateEmptyShopsOverviewData();
const storesOverviewDataForInfiniteScrolling = generateShopsOverviewDataForInfiniteScrolling();
storiesOf("StoresOverview", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add(" implementation without infinite scrolling", () => (
    <StoresOverviewComponent
      filterBar={storesOverviewData.filterBar}
      shopCards={storesOverviewData.shopCards}
      getShopCards={() => ""}
      setShopCards={() => ""}
      totalCards={1}
    />
  ))
  .add("Empty implementation", () => (
    <StoresOverviewComponent
      filterBar={storesOverviewEmptyData.filterBar}
      shopCards={storesOverviewEmptyData.shopCards}
      getShopCards={() => ""}
      setShopCards={() => ""}
      totalCards={1}
    />
  ))
  .add(" implementation with infinite scrolling", () => (
    <StoresOverviewComponent
      filterBar={storesOverviewDataForInfiniteScrolling.filterBar}
      shopCards={storesOverviewDataForInfiniteScrolling.shopCards}
      getShopCards={() => ""}
      setShopCards={() => ""}
      totalCards={1}
    />
  ));
