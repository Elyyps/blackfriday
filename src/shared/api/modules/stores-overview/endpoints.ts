import { getShopCards } from "./stores-overview.module";
import { generateDummyStoresOverview } from "./generate-dummy-data";

export const filterShopsOverviewCards = getShopCards;
export const getShopsOverviewData = generateDummyStoresOverview;
