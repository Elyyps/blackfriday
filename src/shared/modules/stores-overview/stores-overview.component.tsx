import * as React from "react";
import styles from "./stores-overview-component.module.scss";
import { FilterBarComponent } from "@app/core/filter-bar";
import { ShopCardComponent } from "@app/core/shop-card";
import { StoresOverviewContainerProps } from "./container/stores-overview.container";
import { getShopsOverviewData } from "@app/api/modules/stores-overview/endpoints";
import { useEffect, useState } from "react";
import BottomScrollListener from "react-bottom-scroll-listener";
import { ClipLoader } from "react-spinners";
import { css } from "@emotion/core";
import { bannerProps } from "@app/api/core/banner";
import { BannerModuleComponent } from "../banner-module";
import { TabContainerComponent, TabComponent } from "@app/prep/modules-prep/core";
import { tabItems } from "@app/prep/pages-prep/winkleoverview/dummy-data";

export interface IStoresOverviewComponentProps {}

const StoresOverviewComponent = (props: IStoresOverviewComponentProps & StoresOverviewContainerProps) => {
  const [selectedBrands, setSelectedBrands] = useState<string[]>([]);
  const [selectedStatus, setSelectedStatus] = useState<string[]>([]);
  const [selectedCategories, setSelectedCategories] = useState<string[]>([]);
  const [currentPage, setCurrentPage] = useState<number>(0);
  const [isLoading, setIsLoading] = useState<boolean>(false);
  const TAKE = 15;
  let currentNumberOfItems = 0;
  const paddingBottom = "30px";
  const paddingTop = paddingBottom;
  const connectClass = "uk-switcher-list";

  const switcherAttr = { "data-uk-switcher": `connect: .${connectClass}` };
  function bottomPageCallback() {
    if (!isLoading) {
      setIsLoading(true);
      setTimeout(() => {
        setIsLoading(false);
      }, 2000);
      const newCurrentPage = currentPage + 1;
      setCurrentPage(newCurrentPage);
    }
  }
  useEffect(() => {
    props.setShopCards(getShopsOverviewData());
  }, []);

  useEffect(() => {
    props.getShopCards(props.shopCards, currentPage, selectedStatus, selectedCategories, selectedBrands, "");
  }, [selectedStatus]);

  const applyOrderByFilter = (id: string) => {
    props.getShopCards(props.shopCards, currentPage, selectedStatus, selectedCategories, selectedBrands, id);
  };

  const applyFilters = () => {
    props.getShopCards(props.shopCards, currentPage, selectedStatus, selectedCategories, selectedBrands, "");
  };

  useEffect(() => {
    props.getShopCards(props.shopCards, currentPage, selectedStatus, selectedCategories, selectedBrands, "");
  }, [currentPage]);
  const override = css`
    display: block;
    margin: 0 auto;
    border-color: red;
  `;

  return (
    <div className={styles["stores-overview"]}>
      <div
        style={{
          position: "fixed",
          top: 16,
          right: 16,
          color: "white",
          zIndex: 100,
          backgroundColor: "red",
          fontSize: 15
        }}
      />

      <div className="uk-container">
        <div className={styles["stores-overview__header"]}>
          <div className={styles["stores-overview__tab"]}>
            <TabContainerComponent classTabList={styles["uk-tab__list"]}>
              {tabItems
                ? tabItems.map((item, key) => (
                    <TabComponent attrAction={"link"} key={key}>
                      {item.title}
                    </TabComponent>
                  ))
                : ""}
            </TabContainerComponent>
          </div>
          <FilterBarComponent
            filterBar={props.filterBar}
            onBrandChanged={setSelectedBrands}
            onCategoryChanged={setSelectedCategories}
            onStatusChanged={setSelectedStatus}
            onOrderByChanged={applyOrderByFilter}
            applyFilter={applyFilters}
          />
        </div>
        <div className={styles["stores-overview__body"]}>
          {props.shopCards && (
            <div
              // className="uk-grid-posts uk-grid uk-grid-small  uk-child-width-1-3@s uk-child-width-1-4@s uk-child-width-1-5@m"

              className={styles["stores-overview__body__list"]}
            >
              {props.shopCards.map((item, key) => {
                let showAd = false;
                if (currentNumberOfItems + 1 === TAKE) {
                  showAd = true;
                  currentNumberOfItems = 0;
                } else {
                  currentNumberOfItems += 1;
                }

                return (
                  <React.Fragment key={key}>
                    <div className={styles["stores-overview__body__cards"]}>
                      <ShopCardComponent
                        title={item.title}
                        buttonLink={item.button.url}
                        seeMoreText={item.seeMore.title}
                        seeMoreLink={item.seeMore.url}
                        image={item.picture}
                        content={item.content}
                        buttonText={item.button.title}
                        range={item.timeLeftBar.value}
                        subtitle={item.timeLeftBar.text}
                      />
                    </div>

                    {showAd && (
                      <div style={{ width: "100%" }}>
                        <BannerModuleComponent
                          bgcolor="#eee"
                          paddingBottom={paddingBottom}
                          paddingTop={paddingTop}
                          bannerProps={bannerProps}
                        />
                        <br />
                      </div>
                    )}
                  </React.Fragment>
                );
              })}
              <BottomScrollListener onBottom={bottomPageCallback} />
            </div>
          )}
          <div>{props.shopCards.length === 0 && <h1>No results for your search</h1>}</div>
          <div style={{ width: "50px", margin: "auto", paddingTop: "30px" }}>
            <ClipLoader css={override} sizeUnit={"px"} size={30} color={"red"} loading={isLoading} />
          </div>
        </div>
      </div>
    </div>
  );
};

export { StoresOverviewComponent };
