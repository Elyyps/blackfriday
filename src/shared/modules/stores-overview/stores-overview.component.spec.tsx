import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
//import { withProvider } from "@app/modules/core/storybook-provider";
import { StoresOverviewComponent } from "./stores-overview.component";

storiesOf("StoresOverview", module).addDecorator(withA11y);
//.addDecorator(withProvider)
//.add("Basic implementation", () => (
//  <StoresOverviewComponent />
//));
