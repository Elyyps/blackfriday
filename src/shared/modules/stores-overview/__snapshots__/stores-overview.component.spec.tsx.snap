// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[StoresOverview] should render component without crashing 1`] = `
<div
  className="stores-overview"
>
  <div
    style={
      Object {
        "backgroundColor": "red",
        "color": "white",
        "fontSize": 15,
        "position": "fixed",
        "right": 16,
        "top": 16,
        "zIndex": 100,
      }
    }
  />
  <div
    className="uk-container"
  >
    <div
      className="stores-overview__header"
    >
      <FilterBarComponent
        applyFilter={[Function]}
        filterBar={
          Object {
            "brandFilterTitle": "Brands",
            "brands": Array [
              Object {
                "isChecked": false,
                "label": 0,
                "text": "Appel ",
              },
              Object {
                "isChecked": false,
                "label": 255,
                "text": "Adidas",
              },
              Object {
                "isChecked": false,
                "label": 22,
                "text": "Converse",
              },
              Object {
                "isChecked": false,
                "label": 12,
                "text": "Guess",
              },
              Object {
                "isChecked": false,
                "label": 2,
                "text": "JBL",
              },
              Object {
                "isChecked": false,
                "label": 25,
                "text": "Only",
              },
              Object {
                "isChecked": false,
                "label": 0,
                "text": "Test1 ",
              },
              Object {
                "isChecked": false,
                "label": 255,
                "text": "Test2",
              },
              Object {
                "isChecked": false,
                "label": 22,
                "text": "Test3",
              },
              Object {
                "isChecked": false,
                "label": 12,
                "text": "Test4",
              },
              Object {
                "isChecked": false,
                "label": 2,
                "text": "Test5",
              },
              Object {
                "isChecked": false,
                "label": 25,
                "text": "Test6",
              },
              Object {
                "isChecked": false,
                "label": 12,
                "text": "Test7",
              },
              Object {
                "isChecked": false,
                "label": 2,
                "text": "Test8",
              },
              Object {
                "isChecked": false,
                "label": 25,
                "text": "Test9",
              },
            ],
            "categories": Array [
              Object {
                "isChecked": false,
                "label": 12,
                "text": "Audio",
              },
              Object {
                "isChecked": false,
                "label": 3,
                "text": "Electronic",
              },
              Object {
                "isChecked": false,
                "label": 20,
                "text": "Sport",
              },
              Object {
                "isChecked": false,
                "label": 2,
                "text": "Mode",
              },
              Object {
                "isChecked": false,
                "label": 9,
                "text": "Games",
              },
            ],
            "categoryFilterTitle": "Categories",
            "sortBy": Array [
              "Newest",
              "Relevant",
              "Price",
            ],
            "sortByFilterTitle": "Sorted by",
            "status": Array [
              Object {
                "isChecked": false,
                "label": 12,
                "text": "Actie nog onbekend",
              },
              Object {
                "isChecked": false,
                "label": 12,
                "text": "Actie start over 5 dagen!",
              },
              Object {
                "isChecked": false,
                "label": 3,
                "text": "Nu geldig",
              },
              Object {
                "isChecked": false,
                "label": 3,
                "text": "Bijna Verlopen!",
              },
              Object {
                "isChecked": false,
                "label": 3,
                "text": "Actie verlopen",
              },
            ],
            "statusFilterTitle": "Status",
          }
        }
        onBrandChanged={[Function]}
        onCategoryChanged={[Function]}
        onOrderByChanged={[Function]}
        onStatusChanged={[Function]}
      />
    </div>
    <div
      className="stores-overview__body"
    >
      <div
        className="stores-overview__body__list"
      >
        <div
          className="stores-overview__body__cards"
        >
          <ShopCardComponent
            buttonLink=""
            buttonText="Naar deals"
            content="content1"
            image="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/shopCards/1.png"
            range={0.8666193629294792}
            seeMoreLink=""
            seeMoreText="Meer info"
            subtitle="Actie nog onbekend"
            title="Item n: 1"
          />
        </div>
        <div
          className="stores-overview__body__cards"
        >
          <ShopCardComponent
            buttonLink=""
            buttonText="Naar deals"
            content="content2"
            image="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/shopCards/2.png"
            range={0.8656509563559192}
            seeMoreLink=""
            seeMoreText="Meer info"
            subtitle="Actie nog onbekend"
            title="Item n: 2"
          />
        </div>
        <div
          className="stores-overview__body__cards"
        >
          <ShopCardComponent
            buttonLink=""
            buttonText="Naar deals"
            content="content3"
            image="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/shopCards/3.png"
            range={0.2889599294363878}
            seeMoreLink=""
            seeMoreText="Meer info"
            subtitle="Actie nog onbekend"
            title="Item n: 3"
          />
        </div>
        <div
          className="stores-overview__body__cards"
        >
          <ShopCardComponent
            buttonLink=""
            buttonText="Naar deals"
            content="content4"
            image="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/shopCards/4.png"
            range={0.02451662277007549}
            seeMoreLink=""
            seeMoreText="Meer info"
            subtitle="Actie nog onbekend"
            title="Item n: 4"
          />
        </div>
        <div
          className="stores-overview__body__cards"
        >
          <ShopCardComponent
            buttonLink=""
            buttonText="Naar deals"
            content="content5"
            image="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/shopCards/5.png"
            range={0.28620902377285673}
            seeMoreLink=""
            seeMoreText="Meer info"
            subtitle="Actie nog onbekend"
            title="Item n: 5"
          />
        </div>
        <div
          className="stores-overview__body__cards"
        >
          <ShopCardComponent
            buttonLink=""
            buttonText="Naar deals"
            content="content1"
            image="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/shopCards/1.png"
            range={0.859997775444866}
            seeMoreLink=""
            seeMoreText="Meer info"
            subtitle="Actie start over 5 dagen!"
            title="Item n: 1"
          />
        </div>
        <div
          className="stores-overview__body__cards"
        >
          <ShopCardComponent
            buttonLink=""
            buttonText="Naar deals"
            content="content2"
            image="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/shopCards/2.png"
            range={0.04909875783764761}
            seeMoreLink=""
            seeMoreText="Meer info"
            subtitle="Actie start over 5 dagen!"
            title="Item n: 2"
          />
        </div>
        <div
          className="stores-overview__body__cards"
        >
          <ShopCardComponent
            buttonLink=""
            buttonText="Naar deals"
            content="content3"
            image="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/shopCards/3.png"
            range={0.36868934227560635}
            seeMoreLink=""
            seeMoreText="Meer info"
            subtitle="Actie start over 5 dagen!"
            title="Item n: 3"
          />
        </div>
        <div
          className="stores-overview__body__cards"
        >
          <ShopCardComponent
            buttonLink=""
            buttonText="Naar deals"
            content="content4"
            image="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/shopCards/4.png"
            range={0.16381194188113946}
            seeMoreLink=""
            seeMoreText="Meer info"
            subtitle="Actie start over 5 dagen!"
            title="Item n: 4"
          />
        </div>
        <div
          className="stores-overview__body__cards"
        >
          <ShopCardComponent
            buttonLink=""
            buttonText="Naar deals"
            content="content5"
            image="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/shopCards/5.png"
            range={0.32102861022622275}
            seeMoreLink=""
            seeMoreText="Meer info"
            subtitle="Actie start over 5 dagen!"
            title="Item n: 5"
          />
        </div>
        <div
          className="stores-overview__body__cards"
        >
          <ShopCardComponent
            buttonLink=""
            buttonText="Naar deals"
            content="content1"
            image="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/shopCards/1.png"
            range={0.8489169923214028}
            seeMoreLink=""
            seeMoreText="Meer info"
            subtitle="Nu geldig"
            title="Item n: 1"
          />
        </div>
        <div
          className="stores-overview__body__cards"
        >
          <ShopCardComponent
            buttonLink=""
            buttonText="Naar deals"
            content="content2"
            image="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/shopCards/2.png"
            range={0.6279378896278167}
            seeMoreLink=""
            seeMoreText="Meer info"
            subtitle="Nu geldig"
            title="Item n: 2"
          />
        </div>
        <div
          className="stores-overview__body__cards"
        >
          <ShopCardComponent
            buttonLink=""
            buttonText="Naar deals"
            content="content3"
            image="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/shopCards/3.png"
            range={0.2536756648617824}
            seeMoreLink=""
            seeMoreText="Meer info"
            subtitle="Nu geldig"
            title="Item n: 3"
          />
        </div>
        <div
          className="stores-overview__body__cards"
        >
          <ShopCardComponent
            buttonLink=""
            buttonText="Naar deals"
            content="content4"
            image="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/shopCards/4.png"
            range={0.8126324844088779}
            seeMoreLink=""
            seeMoreText="Meer info"
            subtitle="Nu geldig"
            title="Item n: 4"
          />
        </div>
        <div
          className="stores-overview__body__cards"
        >
          <ShopCardComponent
            buttonLink=""
            buttonText="Naar deals"
            content="content5"
            image="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/shopCards/5.png"
            range={0.9899780894140373}
            seeMoreLink=""
            seeMoreText="Meer info"
            subtitle="Nu geldig"
            title="Item n: 5"
          />
        </div>
        <div
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <BannerComponent
            buttonLink="#"
            buttonTitle="Meer deals"
            image={
              Object {
                "alt": "alt text oasis",
                "src": "oasis.jpg",
              }
            }
            label="22 t/m 26 nov."
            logo={
              Object {
                "alt": "alt text CoolBlue",
                "src": "cool-blue.svg",
              }
            }
            text="Een selectie van bizarre aanbiedingen. Meer info"
            title="Black Friday bij Coolblue"
          />
          <br />
        </div>
        <div
          className="stores-overview__body__cards"
        >
          <ShopCardComponent
            buttonLink=""
            buttonText="Naar deals"
            content="content1"
            image="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/shopCards/1.png"
            range={0.43293947171049973}
            seeMoreLink=""
            seeMoreText="Meer info"
            subtitle="Bijna Verlopen!"
            title="Item n: 1"
          />
        </div>
        <div
          className="stores-overview__body__cards"
        >
          <ShopCardComponent
            buttonLink=""
            buttonText="Naar deals"
            content="content2"
            image="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/shopCards/2.png"
            range={0.9401576258816624}
            seeMoreLink=""
            seeMoreText="Meer info"
            subtitle="Bijna Verlopen!"
            title="Item n: 2"
          />
        </div>
        <div
          className="stores-overview__body__cards"
        >
          <ShopCardComponent
            buttonLink=""
            buttonText="Naar deals"
            content="content3"
            image="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/shopCards/3.png"
            range={0.7674763118088384}
            seeMoreLink=""
            seeMoreText="Meer info"
            subtitle="Bijna Verlopen!"
            title="Item n: 3"
          />
        </div>
        <div
          className="stores-overview__body__cards"
        >
          <ShopCardComponent
            buttonLink=""
            buttonText="Naar deals"
            content="content4"
            image="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/shopCards/4.png"
            range={0.0824971515701669}
            seeMoreLink=""
            seeMoreText="Meer info"
            subtitle="Bijna Verlopen!"
            title="Item n: 4"
          />
        </div>
        <div
          className="stores-overview__body__cards"
        >
          <ShopCardComponent
            buttonLink=""
            buttonText="Naar deals"
            content="content5"
            image="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/shopCards/5.png"
            range={0.34094817054965976}
            seeMoreLink=""
            seeMoreText="Meer info"
            subtitle="Bijna Verlopen!"
            title="Item n: 5"
          />
        </div>
        <div
          className="stores-overview__body__cards"
        >
          <ShopCardComponent
            buttonLink=""
            buttonText="Naar deals"
            content="content1"
            image="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/shopCards/1.png"
            range={0.8003016343383624}
            seeMoreLink=""
            seeMoreText="Meer info"
            subtitle="Actie verlopen"
            title="Item n: 1"
          />
        </div>
        <div
          className="stores-overview__body__cards"
        >
          <ShopCardComponent
            buttonLink=""
            buttonText="Naar deals"
            content="content2"
            image="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/shopCards/2.png"
            range={0.28210413829510395}
            seeMoreLink=""
            seeMoreText="Meer info"
            subtitle="Actie verlopen"
            title="Item n: 2"
          />
        </div>
        <div
          className="stores-overview__body__cards"
        >
          <ShopCardComponent
            buttonLink=""
            buttonText="Naar deals"
            content="content3"
            image="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/shopCards/3.png"
            range={0.09670489789464254}
            seeMoreLink=""
            seeMoreText="Meer info"
            subtitle="Actie verlopen"
            title="Item n: 3"
          />
        </div>
        <div
          className="stores-overview__body__cards"
        >
          <ShopCardComponent
            buttonLink=""
            buttonText="Naar deals"
            content="content4"
            image="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/shopCards/4.png"
            range={0.9443337155346603}
            seeMoreLink=""
            seeMoreText="Meer info"
            subtitle="Actie verlopen"
            title="Item n: 4"
          />
        </div>
        <div
          className="stores-overview__body__cards"
        >
          <ShopCardComponent
            buttonLink=""
            buttonText="Naar deals"
            content="content5"
            image="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/shopCards/5.png"
            range={0.7495598550332692}
            seeMoreLink=""
            seeMoreText="Meer info"
            subtitle="Actie verlopen"
            title="Item n: 5"
          />
        </div>
        <BottomScrollListener
          debounce={200}
          offset={0}
          onBottom={[Function]}
        />
      </div>
      <div />
      <div
        style={
          Object {
            "margin": "auto",
            "paddingTop": "30px",
            "width": "50px",
          }
        }
      />
    </div>
  </div>
</div>
`;
